liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java

```diff
@@ -176,6 +179,35 @@ private void migrate(LiquibaseConfigurationProperties config, DataSource dataSou
             }
         }
     }
+    
+          /**
+          * Performs update only when there are unrun changesets. Drastically improves app startup time when there are no 
+          * changes to be performed on database.
+          * @param liquibase Primary facade class for interacting with Liquibase.
+          * @param config    Liquibase configuration
+          * @throws LiquibaseException Liquibase exception.
+          */
+    private void performUpdateIfNeeded(
+        final Liquibase liquibase, LiquibaseConfigurationProperties config)
+            throws LiquibaseException {
+      if (updateNeeded(
+          liquibase, new Contexts(config.getContexts()), new LabelExpression(config.getLabels()))) {
+        ChangeLogHistoryServiceFactory.getInstance()
+        .getChangeLogService(liquibase.getDatabase())
+        .reset();
+        performUpdate(liquibase, config);
+      }
+    }
+
+    private static boolean updateNeeded(
+        final Liquibase liquibase, final Contexts contexts, final LabelExpression labelExpression)
+            throws LiquibaseException {
+      LOG.info("Checking if update required...");
+      final List<ChangeSet> unrunChangeSets =
+          liquibase.listUnrunChangeSets(contexts, labelExpression, false);
+      LOG.info("Size of unrunchangeset : " + unrunChangeSets.size());
```

Comment:
Change to

```java
LOG.debug("Size of unrunchangeset : {}", unrunChangeSets.size());
```

Reply:
Log level changed to debug and removed check style file.